backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala

```diff
@@ -327,48 +325,24 @@ class SparkPlanExecApiImpl extends SparkPlanExecApi {
       child: SparkPlan,
       numOutputRows: SQLMetric,
       dataSize: SQLMetric): BuildSideRelation = {
-    val countsAndBytes = child
-      .executeColumnar()
-      .mapPartitions {
-        iter =>
-          val input = new ArrayBuffer[ColumnarBatch]()
-          // This iter is ClosableColumnarBatch, hasNext will remove it from native batch map,
-          // however, we need collect all batches for serialize, so retain is needed.
-          while (iter.hasNext) {
-            val batch = iter.next
-            if (batch.numCols() != 0) {
-              ColumnarBatches.retain(batch)
-              input += batch
-            }
-          }
-
-          if (input.isEmpty) {
-            Iterator((0L, Array[Byte]()))
-          } else {
-            val handleArray = input.map(ColumnarBatches.getNativeHandle).toArray
-            val serializeResult = ColumnarBatchSerializerJniWrapper
-              .create()
-              .serialize(
-                handleArray,
-                NativeMemoryManagers
-                  .contextInstance("BroadcastRelation")
-                  .getNativeInstanceHandle)
-            input.foreach(ColumnarBatches.release)
-            Iterator((serializeResult.getNumRows, serializeResult.getSerialized))
-          }
-      }
-      .collect
-
-    val batches = countsAndBytes.filter(_._1 != 0).map(_._2)
-    val rawSize = batches.map(_.length).sum
-    if (rawSize >= BroadcastExchangeExec.MAX_BROADCAST_TABLE_BYTES) {
-      throw new SparkException(
-        s"Cannot broadcast the table that is larger than 8GB: ${rawSize >> 30} GB")
+    mode match {
+      case IdentityBroadcastMode =>
+        throw new IllegalStateException("Unreachable code")
+      case HashedRelationBroadcastMode(_, _) =>
+        val serialized: Array[ColumnarBatchSerializeResult] = child
```

Comment:
There is no hash table created in Gluten for the broadcast relation. `IdentityBroadcastMode` and `hashedRelationBroadcastMode` are actually same in case of Gluten.

Please check below comment  in `ColumnarBroadcastExchangeExec`
```
 //this created relation ignore HashedRelationBroadcastMode isNullAware, because we
            // cannot get child output rows, then compare the hash key is null, if not null,
            // compare the isNullAware, so gluten will not generate HashedRelationWithAllNullKeys
            // or EmptyHashedRelation, this difference will cause performance regression in some
            // cases.
```

This match block for mode can be removed safely when BNLJ is supported. Please correct me if my understanding is wrong.

Reply:
The assumption is correct to me. Currently Gluten doesn't handle anything about `IdentityBroadcastMode` so it's reasonable to throw when it sees one. If BNLJ requires the same code with BHJ in this code block, then the one for `IdentityBroadcastMode` can be removed.